### ISSUE TITLE
[FEATURE REQUEST] Use only as a virtual scroller

### ISSUE THREAD

**Issue by ferferga** [rocwang/vue-virtual-scroll-grid#275]
**Is your feature request related to a problem? Please describe.**
I already have all the items loaded when I need to display the grid. However, vue-virtual-scroll-grid doesn't act as a virtual grid only, but rather as a virtual + infinite scroll solution by fetching data while scrolling down.

**Describe the solution you'd like**
An ``items`` prop that take all the loaded items and pass them to the child slots.

**Describe alternatives you've considered**
```
<grid
      v-if="items.length > 0"
      :length="items.length"
      :page-size="items.length"
      :page-provider="async () => items"
      :class="useResponsiveClasses('card-grid-container scroller')">
      <template #probe>
        <card :item="items[0]" margin text overlay link />
      </template>

      <template #default="{ item, style }">
        <card :style="style" :item="item" margin text overlay link />
      </template>
    </grid>
```

With my grid layout, this only renders around 6 columns and 8 rows. The items array passed has 393 items (so that means 48 items are rendered only). The remaining bottom space is calculated correctly (so the scrollbar has the correct sizing) but items simply don't render.

**Comment by rocwang** [comment]
I'm not sure if your description indicates a rendering bug in `vue-virtual-scroll-grid`. However, 
I think you may try slicing your existing 393 items with an arbitrary hardcoded `pageSize`, and then wrap the slice in `Promise.resolve()`  in the `pageprovider` callback:

```js
const items = Array(393).fill('x')

function pageProvicer(pageNumber, pageSize) {
  const slice = items.slice(pageNumber * pageSize, (pageNumber + 1) * pageSize)

  return Promise.resolve(slice)
}
```


```vue
<grid
      v-if="items.length > 0"
      :length="items.length"
      :page-size="20"
      :page-provider="pageProvicer"
      :class="useResponsiveClasses('card-grid-container scroller')"
>
      <template #probe>
        <card :item="items[0]" margin text overlay link />
      </template>

      <template #default="{ item, style }">
        <card :style="style" :item="item" margin text overlay link />
      </template>
</grid>
```

**Comment by ferferga** [comment]
Seems I hit a bug in the rendering, as your solution doesn't work either. The same 48 items are rendered. In one occasion, the virtual scrolling partially worked for me, as scrolling down rendered more items, but scrolling up again left a gap in the top of the container.

I thought [OverlayScrollbars](https://github.com/KingSora/OverlayScrollbars) might be involved in this, but I removed it completely from my project to test and the same thing happened.

I have no problems in pushing and sharing the code [as the project I'm working on is fully public and open source](https://github.com/jellyfin/jellyfin-vue) but I guess that investigating the full source code of a project is your last resort.

All the info that I think it might be useful for you is that I'm using the following classes depending on screen size:

```
.card-grid-container.sm-and-down {
  grid-template-columns: repeat(3, minmax(calc(100% / 3), 1fr));
}

.large-grid.sm-and-down .card-grid-container.sm-and-down {
  grid-template-columns: repeat(2, minmax(calc(100% / 2), 1fr));
}

.card-grid-container.sm-and-up {
  grid-template-columns: repeat(4, minmax(calc(100% / 4), 1fr));
}

.large-grid.sm-and-up .card-grid-container.sm-and-up {
  grid-template-columns: repeat(3, minmax(calc(100% / 3), 1fr));
}

.card-grid-container.lg-and-up {
  grid-template-columns: repeat(6, minmax(calc(100% / 6), 1fr));
}

.large-grid.lg-and-up .card-grid-container.lg-and-up {
  grid-template-columns: repeat(4, minmax(calc(100% / 4), 1fr));
}

.card-grid-container.xl {
  grid-template-columns: repeat(8, minmax(calc(100% / 8), 1fr));
}

.large-grid.xl .card-grid-container.xl {
  grid-template-columns: repeat(5, minmax(calc(100% / 5), 1fr));
}
```

**Comment by rocwang** [comment]
Thanks for the clarification.

Would you mind sharing a git repo with me containing a minimal replications of this issue you are facing?
Then I may take it from there for further investigation.

**Comment by ferferga** [comment]
**Jump into the client**

1. Checkout this branch: https://github.com/jellyfin/jellyfin-vue/tree/vite
2. Start the client: ``npm i && npm start``. It will listen in 127.0.0.1:3000
3. When you're asked for server info, type ``https://demo.jellyfin.org/stable``
4. Type ``demo`` and leave empty password (this is also explained in the login instructions)

** Structure**
Your component is used in ``ItemGrid.vue``, located at ``frontend/src/components/Item/ItemGrid.vue``
The parent component is located at ``frontend/src/pages/library/_viewId.vue``

**How to test**

The demo server we have doesn't have enough items to properly test the scroller. The **Movies** library is the one that has the most items, so it's the best one to test the scroller: you can access it by clicking "Movies" in the hamburger menu located at the left, but you're free to switch between them.

In order to solve the lack of enough items, you can manipulate the array and duplicate them as you wish [in this line](https://github.com/jellyfin/jellyfin-vue/blob/vite/frontend/src/pages/library/_viewId.vue#L298).

Another useful resource is at ``frontend/src/composables/use-responsive-classes.ts`` composable, which is called in ``ItemGrid.vue``'s template to generate responsive CSS classes for each layout according to Vuetify's breakpoints (look at the scoped styles of ``ItemGrid`` and see the CSS selectors I use).

The repo is also codespace friendly, so it should be easy to get onboard if you don't want to clone into your computer.

Hopefully this is enough to help you diagnose it, but of course don't hesitate to contact with me if you face any problem.

Thank you very much for your best effort in helping me with this because, although I'm looking forward to contribute to this project (I took a look at the code and I think some things can be removed from rxjs, rendering a much smaller bundle size), I want to have our Vue 3 migration complete before focusing on anything else, so get a workeable thing with this lib is my top priority right now.

**Comment by rocwang** [comment]
@ferferga 

 I replicated the bug in your project and fixed it in v1.10.1. Please give it a try and let me know if it works for you.

**Comment by ferferga** [comment]
@rocwang Upgrading indeed fixed the issue.

Thank you very much!
